aws_lambda_powertools/logging/types.py

```diff
@@ -0,0 +1,10 @@
+from typing import Any, Dict, Iterable, Union
+
+from typing_extensions import TypedDict
+
+class TypedLog(TypedDict, total=False):
```

Comment:
Extend your TypedDict class to include optional fields when customers choose to inject Lambda context, correlation ID, etc. - our documentation has examples of those fields.

Here's a complete example to make it easier and when they're added for context:

```python

from typing_extensions import NotRequired

class PowertoolsLogRecord(TypedDict):
    level: str
    location: str
    timestamp: str | int
    service: str
    message: str | bool | Dict[str, Any] | List[Any]
    cold_start: NotRequired[bool]  # logger.inject_lambda_context
    function_name: NotRequired[str]  # logger.inject_lambda_context
    function_memory_size: NotRequired[int]  # logger.inject_lambda_context
    function_arn: NotRequired[str]  # logger.inject_lambda_context
    function_request_id: NotRequired[str]  # logger.inject_lambda_context
    xray_trace_id: NotRequired[str]  # logger.inject_lambda_context if AWS X-Ray is enabled
    correlation_id: NotRequired[str]  # logger.set_correlation_id
    exception_name: NotRequired[str]  # logger.exception
    exception: NotRequired[str]  # logger.exception
```


**Caveat**

You need to adjust your import to make it conditional for both `NotRequired` and `TypedDict`. The former (NotRequired) is only supported in 3.11 so import from `typing_extensions` would fail in 3.11, and `TypedDict` is available in 3.8+ and it'd also fail later.

> Docs: https://docs.python.org/3/library/typing.html

I'll let you figure out how to make an conditional import depending on the Python version, but please shout if you get stuck!